src/Worker.cpp

```diff
@@ -207,19 +237,28 @@ void Worker::refill_buffer(int recv_len) {
         received_bytes = recv(reader_fd,get_unread_buffer() + get_bmp_data_unread_len(),
                 recv_len,MSG_WAITALL);
         if (received_bytes <= 0) {
-            LOG_INFO("bad connection");
+            LOG_INFO("bad connection from router: %d, stopping worker", router_ip.c_str());
             // set worker status to stopped. the main thread will clean up later.
             status = WORKER_STATUS_STOPPED;
         }
         bmp_data_unread_len += received_bytes;
+    } else {
+        LOG_ERR("Buffer overflow in worker! router: %d", router_ip.c_str());
+        status = WORKER_STATUS_STOPPED;
     }
 
 }
 
 void Worker::update_buffer(int parsed_bmp_msg_len) {
     bmp_data_read_len += parsed_bmp_msg_len;
     bmp_data_unread_len -= parsed_bmp_msg_len;
-    assert(bmp_data_read_len >= 0);
+    if(bmp_data_read_len < 0){
+        LOG_ERR("Assertion error. bmp_data_read_len is invalid (%d)", bmp_data_read_len);
+    }
+    assert(bmp_data_unread_len >= 0);
+    if(bmp_data_unread_len < 0){
+        LOG_ERR("Assertion error. bmp_data_unread_len is invalid (%d)", bmp_data_read_len);
```

Comment:
Copy paste typo -- want to print `unread`

Reply:
Fixed.